### PR TITLE
*: remove hot region check (#90)

### LINKED ISSUE #5535: SplitAndScatter will not work when region is hot
### What did you do?

BR uses SplitAndScatter gRPC interface of PD to split region and scatter regions.

### What did you expect to see?

Split and scatter successfully.

### What did you see instead?

add log
img_v2_aaab6fe9-7d77-495d-8761-feb021cfa89g
Then see
img_v2_2797fe4c-fd99-40c6-b2cb-bcc14258810g

### What version of PD are you using (`pd-server -V`)?
master

### PR DESCRIPTION
**Pull request by rleungx**
### What problem does this PR solve?

Issue Number: Close #5535.

### What is changed and how does it work?

Skip the hot region check when splitting. 
cherry-pick https://github.com/tidbcloud/pd-cse/pull/90.

```commit-message
```

### Check List

Tests

- Unit test

### Release note

```release-note
None.
```

### CODE REVIEW

File: pkg/schedule/splitter/region_splitter.go [hunk 1]
```diff
@@ -147,6 +147,9 @@ func (r *RegionSplitter) groupKeysByRegion(keys [][]byte) map[uint64]*regionGrou
 		if !r.checkRegionValid(region) {
 			continue
 		}
+		if bytes.Equal(region.GetStartKey(), key) {
```
Comment (lhy1024) [inline pkg/schedule/splitter/region_splitter.go:150]: Do we need to check end key?

Comment (rleungx) [reply pkg/schedule/splitter/region_splitter.go:150]: No, we don't need it.

Comment (CabinfeverB) [reply pkg/schedule/splitter/region_splitter.go:150]: If key is strart key of a region, does TiKV report error?

**Review by CabinfeverB — COMMENTED** [review]
LGTM. but I find scatter will check whether region is hot too, do we need also skip?

**Comment by rleungx** [comment]
> LGTM. but I find scatter will check whether region is hot too, do we need also skip?

IMO, it's not related to SplitAndScatter, maybe we can skip it after we meet the problems.

**Review by CabinfeverB — APPROVED** [review]

**Pull request merged**